### PR TITLE
Add uniform buffer subtests exposing ANGLE bugs

### PR DESCRIPTION
**Pull request by Oletus**
These tests pass on ANGLE's GL backend but failed on the DX backend
until very recent fixes.

### CODE REVIEW

File: sdk/tests/conformance2/buffers/uniform-buffers.html [hunk 1]
```diff
@@ -114,6 +114,58 @@
...
+
+struct wrapper_t {
+    color_t color;
+};
+
+uniform UBOData {
+    wrapper_t UBOStruct;
+};
+
+// This is intended to reproduce a specific ANGLE bug that triggers when the wrapper struct is passed to a function.
```
Comment (zhenyao) [inline sdk/tests/conformance2/buffers/uniform-buffers.html]: Could you add a reference link to the ANGLE bug here?

File: sdk/tests/conformance2/buffers/uniform-buffers.html [hunk 2]
```diff
@@ -114,6 +114,58 @@
...
+    float red;
+    float green;
+    float blue;
+};
+
+uniform UBOData {
+    color_t UBOColors[2];
+};
+
+// This is intended to reproduce a specific ANGLE bug that triggers when a struct from an array of structs in an interface block is passed to a function.
```
Comment (zhenyao) [inline sdk/tests/conformance2/buffers/uniform-buffers.html]: Could you add a reference link to the ANGLE bug here?

**Comment by zhenyao** [comment]
LGTM with a request for linking to the ANGLE bug.
Feel free to merge yourself once it's added.

**Comment by Oletus** [comment]
Added the bug links, merging. Thanks for the reviews!

**Pull request merged**